nucliadb/nucliadb/ingest/orm/processor/__init__.py

```diff
@@ -429,16 +448,18 @@ async def txn(
                 else:
                     raise AttributeError("Shard is not available")
 
-                await txn.commit(partition, seqid)
+                if transaction_check:
+                    await sequence_manager.set_last_seqid(txn, partition, seqid)
+                await txn.commit()
 
                 # Slug may have conflicts as its not partitioned properly. We make it as short as possible
-                txn = await self.driver.begin()
-                resource.txn = txn
-                await resource.set_slug()
-                await txn.commit(resource=False)
+                # XXX Does this write need to happen every time?
+                async with self.driver.transaction() as txn:
```

Comment:
I think we should name this transaction differently to avoid leaving the main one open?

Reply:
The other txn has already been committed at this point but good practice anyways!